.github/workflows/policy_controller.yml

```diff
@@ -120,3 +123,122 @@ jobs:
           done
 
           exit $ex
+
+  docker_build:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        component:
+          - controller
+          - policy-controller
+          - proxy
+    name: Docker build (${{ matrix.component }})
+    timeout-minutes: 30
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@a12a3943b4bdde767164f792f33f40b04645d846
+      - uses: ./.github/actions/docker-build
+        with:
+          docker-registry: ghcr.io/linkerd
+          docker-target: linux-amd64
+          component: ${{ matrix.component }}
+      - name: Create artifact with CLI and image archives
+        run: |
+          mkdir -p /home/runner/archives
+          docker save "ghcr.io/linkerd/${{ matrix.component }}:$TAG" \
+            >/home/runner/archives/${{ matrix.component }}.tar
+      - name: Upload artifact
+        uses: actions/upload-artifact@6673cd052c4cd6fcf4b4e6e60ea986c889389535
+        with:
+          name: image-archives
+          path: /home/runner/archives
+
+  integration:
+    needs: [docker_build]
+    name: Policy controller integration
+    runs-on: ubuntu-20.04
+    timeout-minutes: 20
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@a12a3943b4bdde767164f792f33f40b04645d846
+
+      - name: Install kubectl
+        run: |
+          dir=$(mktemp -d)
+          version=$(bin/scurl https://dl.k8s.io/release/stable.txt)
+          bin/scurl -vo $dir/kubectl "https://dl.k8s.io/release/${version}/bin/linux/amd64/kubectl"
+          chmod 755 $dir/kubectl
+          sudo mv $dir/kubectl /usr/local/bin/kubectl
+          cli_version=$(kubectl version --client --short | sed -ne 's/^Client Version: //p')
+          if [[ "$version" != "$cli_version" ]] ; then
+            echo "::error::kubectl version is $cli_version; expected $version"
+            exit 1
+          fi
+          kubectl version --client
+      - name: Install k3d
+        run: |
+          bin/scurl -v https://raw.githubusercontent.com/rancher/k3d/main/install.sh | bash
+          k3d --version
```

Comment:
Not opposed to using k3d-action for more complex uses, but the current approach is actually pretty simple as-is:

```yaml
      - run: bin/scurl -v https://raw.githubusercontent.com/k3d-io/k3d/${K3D_VERSION}/install.sh | bash
      - run: k3d --version
      - run: k3d cluster create --no-lb --k3s-arg "--no-deploy=local-storage,traefik,servicelb,metrics-server@server:*"
      - run: kubectl version
```

One small downside to k3d-action is that we have to specify a cluster name rather than just using the default